File: typescript-javascript/algorithms/easy/plusMinus.ts
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-05-19
 * Last modified: 2026-06-11
 * @description Plus Minus - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/plus-minus/problem}
 */
//---------- functions ----------
/**
 * Calculates and prints the ratios of positive, negative, and zero elements in an array.
 * @param {number[]} arr - The array of integers.
 */
function plusMinus(arr: number[]): void 
{
    let plusList = arr.filter((el) => el > 0);
    let minusList = arr.filter((el) => el < 0);
    let zerosList = arr.filter((el) => el === 0);
    
    console.log(plusList.length/arr.length);
    console.log(minusList.length/arr.length);
    console.log(zerosList.length/arr.length);
}//end plusMinus()

File: go/algorithms/easy/plusMinus.go
/*
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-13
Last modified: 2026-06-13
Description: Plus Minus - HackerRank
See: https://www.hackerrank.com/challenges/plus-minus/problem
*/
//--------- package ---------
package easy


//--------- Imports ---------
import "fmt"


//--------- functions ---------
// plusMinus Calculates and prints the ratios of positive, negative, and zero elements in an array.
func plusMinus(arr []int32) {
	var plusCount, minusCount, zeroCount int32 = 0, 0, 0;
	
	for _, value := range arr {
		if value > 0 {
			plusCount++;
		} else if value < 0 {
			minusCount++;
		} else {
			zeroCount++;
		}
	}//end for
	
	fmt.Printf("%.6f\n", float64(plusCount)/float64(len(arr)))
	fmt.Printf("%.6f\n", float64(minusCount)/float64(len(arr)))
	fmt.Printf("%.6f\n", float64(zeroCount)/float64(len(arr)))
}//end plusMinus()